addons/mail/models/mail_thread.py

```diff
@@ -2455,9 +2456,11 @@ def _notify_thread_by_email(self, message, recipients_data, msg_vals=False,
             force_email_company=force_email_company,
             force_email_lang=force_email_lang,
         ) # 10 queries
+        if subtitles:
+            template_values['subtitles'] = subtitles
```

Comment:
Shouldn't we propagate it until ``_notify_by_email_prepare_rendering_context`` with ``force_email_subtitles`` parameters, so that this method always returns a complete rendering context ? A bit like force_email_company and force_email_lang.

Reply:
The usual use-case is that the document model overrides the method _notify_by_email_prepare_rendering_context and set the subtitles based on record information.
But for activity notification, it is not possible because _notify_by_email_prepare_rendering_context is called on the document related to the  activity (ex.: Activity: Schedule call, Document: Lead) and the subtitles must be based on activity information and not on document information.
So I have added a parameter force_subtitles in _notify_thread_by_email that can override what could be set in _notify_by_email_prepare_rendering_context. That parameters is being propagated through kwargs from message_notify -> _notify_thread -> _notify_thread_by_email.
Finally that parameters has been renamed subtitles (instead of force_subtitles) after discussing it above in this review.

If we propagate it down to _notify_by_email_prepare_rendering_context, all overriding of that method would have to take care of that parameter. Instead here, we simply override the values returned by _notify_by_email_prepare_rendering_context with "force_subtitles" (subtitles now) if it is set in the main method mail_thread._notify_thread_by_email.

With that explanation in mind, do you think it is better to propagate it down to _notify_by_email_prepare_rendering_context ?

Reply:
"If we propagate it down to _notify_by_email_prepare_rendering_context, all overriding of that method would have to take care of that parameter."

Good remark ... well that's ok then, I suppose both approaches have their pros and cons, so let's keep what you did :)